Lab-4/kalmroth_lab4.ipynb

```diff
@@ -0,0 +1,286 @@
+    "kernelspec": {
+      "name": "python3",
+      "display_name": "Python 3"
+    }
+  },
+  "cells": [
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "6RgtXlfYO_i7",
+        "colab_type": "text"
+      },
+      "source": [
+        "# **Lab 4: Function approximation**\n",
+        "**Kristoffer Almroth**"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "9x_J5FVuPzbm",
+        "colab_type": "text"
+      },
+      "source": [
+        "# **Abstract**"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "6UFTSzW7P8kL",
+        "colab_type": "text"
+      },
+      "source": [
+        "Fourth lab in the course DD2363 Methods in Scientific Computing. This lab is about function approximations."
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "28xLGz8JX3Hh",
+        "colab_type": "text"
+      },
+      "source": [
+        "# **Set up environment**"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "D2PYNusD08Wa",
+        "colab_type": "text"
+      },
+      "source": [
+        "Dependencies needed for running the code."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "Xw7VlErAX7NS",
+        "colab_type": "code",
+        "colab": {}
+      },
+      "source": [
+        "# Load neccessary modules.\n",
+        "from google.colab import files\n",
+        "\n",
+        "import time\n",
+        "import numpy as np\n",
+        "import sympy as sp\n",
+        "from sympy import *\n",
+        "import random\n",
+        "\n",
+        "from matplotlib import pyplot as plt\n",
+        "from matplotlib import tri\n",
+        "from matplotlib import axes\n",
+        "from mpl_toolkits.mplot3d import Axes3D\n",
+        "\n",
+        "import unittest"
+      ],
+      "execution_count": 0,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "gnO3lhAigLev",
+        "colab_type": "text"
+      },
+      "source": [
+        "# **Introduction**"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "l5zMzgPlRAF6",
+        "colab_type": "text"
+      },
+      "source": [
+        "Function approximations is an important tool. Here we make an $L_2$ projection onto the the function space $V_{h}^{1}$ where $V_{h}^{1}$ is an continuous piecewise polynomial space given by \n",
+        "\n",
+        "$P_hf(x) = \\sum_{j=0}^{m+1} \\alpha_j \\phi_j (x)$\n",
+        "\n",
+        "We want to solve the equation $Ax=b$ where \n",
+        "\n",
+        "$a_{ij} = (\\phi_j, \\phi_i)$\n",
+        "\n",
+        "$b_i=(f, \\phi_i)$"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "WeFO9QMeUOAu",
+        "colab_type": "text"
+      },
+      "source": [
+        "# **Methods**"
```

Comment:
Add either an explanation to how you got the different values for A and b, or include a reference to where these were found.